packages/flutter/test/painting/image_provider_test.dart

```diff
@@ -254,10 +222,135 @@ void main() {
           expect(events[i].expectedTotalBytes, kTransparentImage.length);
         }
       }, skip: isBrowser);
+
+      test('Uses http request headers', () async {
+        debugNetworkImageHttpClientProvider = respondOnAnyWithHeaders;
+
+        final Completer<bool> imageAvailable = Completer<bool>();
+        final ImageProvider imageProvider = NetworkImage(nonconst('foo'),
+          headers: const <String, String>{'flutter': 'flutter'},
+        );
+        final ImageStream result = imageProvider.resolve(ImageConfiguration.empty);
+        result.addListener(ImageStreamListener(
+              (ImageInfo image, bool synchronousCall) {
+            imageAvailable.complete(true);
+          },
+          onError: (dynamic error, StackTrace stackTrace) {
+            imageAvailable.completeError(error, stackTrace);
+          },
+        ));
+        expect(await imageAvailable.future, isTrue);
+      }, skip: isBrowser);
     });
   });
 }
 
 class MockHttpClient extends Mock implements HttpClient {}
 class MockHttpClientRequest extends Mock implements HttpClientRequest {}
 class MockHttpClientResponse extends Mock implements HttpClientResponse {}
+class MockHttpHeaders extends Mock implements HttpHeaders {}
+
+HttpClient throwOnAnyClient1() {
```

Comment:
We should add a test that simulates an error in the HttpClientResponse stream while loading a NetworkImage.

Reply:
Done